src/tsdb.c

```diff
@@ -224,85 +225,128 @@ int SeriesAddSample(Series *series, api_timestamp_t timestamp, double value) {
     return TSDB_OK;
 }
 
-int SeriesQuery(Series *series, SeriesIterator *iter, 
-                            api_timestamp_t minTimestamp, api_timestamp_t maxTimestamp) {
+// Initiates SeriesIterator, find the correct chunk and initiate a ChunkIterator
+int SeriesQuery(Series *series, SeriesIterator *iter) {
     timestamp_t rax_key;
     iter->series = series;
     ChunkFuncs *funcs = iter->series->funcs;
     
-    // get the rightmost chunk whose base timestamp is smaller or equal to minTimestamp
-    seriesEncodeTimestamp(&rax_key, minTimestamp);
-    iter->dictIter = RedisModule_DictIteratorStartC(series->chunks, "<=", &rax_key, sizeof(rax_key));
-    RedisModule_DictNextC(iter->dictIter, NULL, (void*)&iter->currentChunk);
-    
-    // iterate to the first relevant chunk
+    seriesEncodeTimestamp(&rax_key, iter->minTimestamp);
+
     void *dictResult = (void *)TRUE;
-    while (dictResult) {
-        if (funcs->GetLastTimestamp(iter->currentChunk) < minTimestamp) {
-            dictResult = RedisModule_DictNextC(iter->dictIter, NULL, (void*)&iter->currentChunk);
-        } else {
-            iter->chunkIterator = funcs->NewChunkIterator(iter->currentChunk);
-            iter->minTimestamp = minTimestamp;
-            iter->maxTimestamp = maxTimestamp;
-            return REDISMODULE_OK;
-        }
+    if (iter->reverse == NO_OPT) {
+        iter->dictIter = RedisModule_DictIteratorStartC(series->chunks, "<=", &rax_key, sizeof(rax_key));
+        RedisModule_DictNextC(iter->dictIter, NULL, (void*)&iter->currentChunk);
+        while (dictResult) {
+            if (funcs->GetLastTimestamp(iter->currentChunk) < iter->minTimestamp) {
+                dictResult = RedisModule_DictNextC(iter->dictIter, NULL, (void*)&iter->currentChunk);
+            } else break;
+        }        
+    } else {    
+        iter->dictIter = RedisModule_DictIteratorStartC(series->chunks, "$", &rax_key, sizeof(rax_key));
```

Comment:
@ashtul I think its worthing mentioning why we start from the end and not look up for the first chunk.